argnorm/drug_categorization.py

```diff
@@ -16,20 +38,29 @@ def confers_resistance_to(aro_num: str) -> List[str]:
         target (list[str]):
             A list with ARO number of the drugs/antibiotics to which the input gene confers resistance to.
     '''
-
+    antibiotic_molecule_node = [ARO['ARO:1000003'], ARO['ARO:1000001']]
+    # some gene superclasses can map to drugs which are immediate children of 'antibiotic molecule'
+    # only use these if no other drugs can be found, as this information will be present in
+    # drugs to drug classes
+    backup_drugs = []
     target = set()
-
     for term in ARO[aro_num].superclasses():
         for drug in term.relationships.get(confers_resistance_to_drug_class_rel, []):
-            target.add(drug.id)
+            if list(ARO[drug.id].superclasses())[1:] == antibiotic_molecule_node:
+                backup_drugs.append(drug.id)
+            else:
+                target.add(drug.id)
 
         for drug in term.relationships.get(confers_resistance_to_antibiotic_rel, []):
-            target.add(drug.id)
+            if list(ARO[drug.id].superclasses())[1:] == antibiotic_molecule_node:
+                backup_drugs.append(drug.id)
+            else:
+                target.add(drug.id)
 
-        if target:
-            break
+    if target == set():
```

Comment:
`if target == set()` is the same (in this context) as `if target`

Reply:
I only want to add the backup drugs if `target` is an empty set. I wanted to make sure that only if some gene superclasses map to drug classes which are immediate children of `antibiotic molecule` and no other drugs can be found we would include those drug classes as outputs of `confers_resistance_to()`. This is because these drug classes will be returned in the `drugs_to_drug_classes()` function. `if target` wouldn't be able to check if `target` is actually an empty set and would add the backup drugs if there were already drugs in it, right? This would mean the drug classes are already reported in the drugs column.

Reply:
Sorry, it should have been `if not target`

Reply:
Got it, I'll use that then